### PR TITLE
feat: add options in CLI to enable UNIX sockets

### PR DESCRIPTION
**Pull request by Xowap**
Use case: my automated deployment system only deals with UNIX sockets which are put behind a Nginx front-end and a domain name. This is because otherwise I'd have to do a lot of bookkeeping about which process uses which port and so forth.

Since I need to ship my app I did this quickly based on the 1.x branch (the dev branch doesn't work on my project) so I guess there's some rebase to do. I also guess that some tests might be in order, although I'm not sure how to bring up automated tests for this.

Please tell me if this feature is desirable and if so what more you expect before merging it!

Thanks,
Rémy

### CODE REVIEW

File: lib/core/nuxt.js [hunk 1]
```diff
@@ -136,23 +136,41 @@ module.exports = class Nuxt {
...
-          // eslint-disable-next-line no-console
-          console.log(
-            '\n' +
-              chalk.bgGreen.black(' OPEN ') +
-              chalk.green(` http://${_host}:${port}\n`)
-          )
+          if (!unix) {
+            const _host = host === '0.0.0.0' ? 'localhost' : host
+            // eslint-disable-next-line no-console
+            console.log(
```
Comment (galvez) [inline lib/core/nuxt.js]: Please use [consola](https://github.com/nuxt/consola) for these calls.

**Review by galvez — CHANGES_REQUESTED** [review]
@Atinux @clarkdo any objections? LGTM except for the logging idiom.

**Comment by galvez** [comment]
@Xowap seems there are some conflicts and tests left to figure out.

File: bin/nuxt-start [hunk 2]
```diff
@@ -38,6 +39,7 @@ if (argv.help) {
     Options
       --port, -p            A port number on which to start the application
       --hostname, -H        Hostname on which to start the application
+      --unix-socket, -u     Path to a UNIX socket
```
Comment (pi0) [inline bin/nuxt-start]: Shouldn't be --unix-socket, -n ?

File: bin/common/utils.js [hunk 3]
```diff
@@ -38,6 +38,10 @@ exports.getLatestHost = argv => {
     argv.hostname ||
     process.env.HOST ||
     process.env.npm_package_config_nuxt_host
+  const unix =
+    argv['unix-socket'] ||
+    process.env.UNIX_SOCKET ||
+    process.env.npm_package_config_unix_socket
 
-  return { port, host }
+  return { port, host, unix }
```
Comment (pi0) [inline bin/common/utils.js]: I would suggest `socket` for variable name. Unix is kind of not exactly related to sockets

File: lib/core/nuxt.js [hunk 4]
```diff
@@ -136,23 +136,41 @@ module.exports = class Nuxt {
...
+              '\n' +
+                chalk.bgGreen.black(' OPEN ') +
+                chalk.green(` http://${_host}:${port}\n`)
+            )
+          } else {
+            // eslint-disable-next-line no-console
+            console.log(
+              '\n' +
+                chalk.bgGreen.black(' OPEN ') +
```
Comment (pi0) [inline lib/core/nuxt.js]: OPEN is not an appreciate badge. Maybe `LISTENING`?

File: lib/core/nuxt.js (hunk 1 above)
Comment (qm3ster) [reply lib/core/nuxt.js]: :b:onzola

**Comment by galvez** [comment]
There's also some level of conflict with https://github.com/nuxt/nuxt.js/pull/3403

**Comment by galvez** [comment]
Closing this as I'll push a separate forked branch from this that integrates with https://github.com/nuxt/nuxt.js/pull/3403.

**Pull request closed without merging**